### ISSUE TITLE
ConnectEx tcp: No connection could be made because the target machine actively refused it

### ISSUE THREAD

**Issue by nathanleclaire** [docker/machine#1094]
Sorry to bring this up on the weekend, I just want to note it down for later reference.

I'm seeing a lot of errors trying to create a machine on my Windows 7 computer at home (this is master branch, probably `03c1fbadc5f53fd863d4149b5b89c9f5bf8e066d`):

![image](https://cloud.githubusercontent.com/assets/1476820/7442392/c9eea9d4-f0c4-11e4-8057-ce687ac85600.png)

EDIT: Ah, I wasn't thinking when I wrote this, but it looks more like a daemon connection issue due to the port 2376 being used.

cc @ehazlett @sthulb

**Comment by ehazlett** [comment]
Yeah looks like daemon isn't running.  I've tested this on Windows 8.1.  Perhaps we should set a min version to run with?

/cc @ahmetalpbalkan

**Comment by ahmetalpbalkan** [comment]
@ehazlett I'm not sure if there are any reasons that it wouldn't run on Win 7 as well. I've seen this error on on OS X as well, for me it was just vbox flakiness, which might very well be happening on Windows as well. Instead of setting a min version, I would try to find the root cause, because it's probably the case in newer Windows versions as well if it's something depending on the setup etc.

**Comment by nathanleclaire** [comment]
Yeah, it's definitely possible it might be something weird in my VBox setup on that computer / Win7.  Maybe I can track down another computer with Win7 on it and try to duplicate, and/or try latest master.

**Comment by ehazlett** [comment]
cool thanks!

**Comment by nathanleclaire** [comment]
Haven't been able to dupe this again

**Comment by nathanleclaire** [comment]
Reopening this.  Kitematic is duplicating this with 0.3.0 and some users are running into it.  FYI @jeffdm maybe you can provide a little more info about the current situation.

**Comment by nathanchere** [comment]
Not scientific by any means but I have had the issue on two machines which had VirtualBox already installed, no issue on two boxes where VirtualBox was installed as part of docker-install.exe.
All Windows 7 x64.

**Comment by pizzurro** [comment]
Got this in Windows 8.1 -- VirtualBox installed prior to docker-machine. 

Executed:
`docker run ubuntu:14.04 /bin/echo 'Hello world'`

Received:
```
Post http://127.0.0.1:2375/v1.19/containers/create: dial tcp 127.0.0.1:2375: ConnectEx: No connection could be made because the target machine actively refused it.. Are you trying to connect to a TLS-enabled daemon without TLS?
```

**Comment by leomrocha** [comment]
Same issue, 
Running as administrator I get:
![Uploading 2015-07-23 10_18_34-Administrator_ Windows PowerShell ISE.png…]()

Running as normal user I get:

![Uploading 2015-07-23 10_19_58-Windows PowerShell ISE.png…]()

Note that environment variables are OK (yesterday I managed to make it work, but today it stopped working after boot2docker update) and that Set-ExecutionPolicy Unrestricted was executed.

My VBox was installed by hand before having docker installed

Note also that I can connect with boot2docker ssh to the VM and then execute things.

**Comment by smarques** [comment]
:+1:

**Comment by PumpkinSeed** [comment]
Same problem there +1

**Comment by nathanchere** [comment]
I haven't dug too far because I simply don't care as long as it works but here is my anecdotal evidence for what is happening.

With me it appears to have been a VirtualBox-related problem, not something specific to Docker at all. On the machines where I installed VirtualBox myself it was running version 5-point-something. It turns out as of version 4.3.12 there were changes made to VirtualBox where it refuses to start if it fails a 'hardening' check on start-up. Amongst other things that check involves an integrity check of the host. In my instance the check failed because I was using a lot of modified files for theming purposes (e.g. replacing icons in shell32.dll).

The Docker-provisioned VirtualBox was installing an older version of VirtualBox so I wasn't encountering the same issue. Two possible solutions:

* Use something like `/sfc scannow` to revert modified files to their known 'safe' state; or
* Install a version of VirtualBox no more recent than 4.3.12

**Comment by dgageot** [comment]
Can @everybody test with docker-machine 0.5.6?

**Comment by frankschmitt** [comment]
I'm still getting this with docker-machine 0.5.6 (Windows 7 Enterprise, 64bit).

For me, the root cause seems to be my corporate VPN client (Juniper Network Connect).
Whenever I establish a VPN connection, it apparently disables the NIC that VirtualBox uses to communicate with the DockerHost VM. When I shutdown the VPN client, everything starts working again properly.

**Comment by dipaktelangre** [comment]
I started getting this issue after docker update

**Comment by Vespira** [comment]
Same here,  we got this issue on Win 10 & Win 7 64bits. It have to be something blocking the connection to Docker engine over the port 2376. But I don't see this port being used, neither I am running some kind of local firewall, etc ... It works when I disable the TLS Verification for Daemon on the Docker machine, but hey, I really want to solve the root issue of this. Any update ?

**Comment by loza0269** [comment]
I'm getting the same error on Windows 7 enterprise. The only thing I can think of is that SEP is blocking it. Could that be possible?

**Comment by SKKOVVURI** [comment]
+1 
I am using docker-machine 0.13.0 and docker 18.01.0-ce
I have the same issue and even I suspect SEP (Symatec Enpoint Protection - Firewall and Antivirus) blocking it.. I stopped it from starting its services and restarted my system. But still I have the same issue. Will try to reinstall docker and virtual box without starting SEP. 
For now, anything I try ends up with t following error msg:

_(default) DBG | ip addr show
(default) DBG | Error dialing TCP: dial tcp 127.0.0.1:22: connectex: No connection could be made because the target machine actively refused it.
(default) DBG | Error dialing TCP: dial tcp 127.0.0.1:22: connectex: No connection could be made because the target machine actively refused it.
(default) DBG | Error dialing TCP: dial tcp 127.0.0.1:22: connectex: No connection could be made because the target machine actively refused it.
(default) DBG | Error dialing TCP: dial tcp 127.0.0.1:22: connectex: No connection could be made because the target machine actively refused it._

If I open VirtualBox and try running docker from inside the default VM, I get

_docker: Error response from daemon, Get https://registry-1.docker.io/v2/:6 net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)._

Thanks,
SK

**Comment by SKKOVVURI** [comment]
I was able to resolve one piece. I just changed NAT to "Bridged" and restarted the VM. so I am able to pull and run images from inside the VM. 
Still have the issue from connecting docker-machine (ssh) to the default VM.

127.0.0.1:22: connectex: No connection could be made because the target machine actively refused it.

BTW, I am running on Win 7... Any thoughts??

**Comment by hero0926** [comment]
Still happening on Win7, Redis on docker is running and could check on cli, but can't connect with gui or api.

**Comment by dataspun** [comment]
Happening on Windows 10 64-bit
Docker version 17.12.0-ce, build c97c6d6
docker-machine version 0.13.0, build 9ba6da9

**Comment by MikyChow** [comment]
+1,
i am using docker machine 14.0 and docker 18.03.0-ce on windows 10 64bit
does any know how to resolve it?
[log.txt](https://github.com/docker/machine/files/1943194/docker-machine.log.txt)

**Comment by waqar-husain** [comment]
this error is on wampp , when i tried to use ngrok free ssl for localhost, on windows 10. not found a fix yet

**Comment by palade** [comment]
> 
> 
> Happening on Windows 10 64-bit
> Docker version 17.12.0-ce, build c97c6d6
> docker-machine version 0.13.0, build [9ba6da9](https://github.com/docker/machine/commit/9ba6da9ebd1140b65eb73e1ce08086ca72764c60)
> 
> Any workaround, fix or troubleshooting tips?

same problem here

**Comment by simonm3** [comment]
I have same problem windows 10 64bit, DockerVersion 2.0.0.2 (30215)

**Comment by cenelsen** [comment]
Same problem here with 64-bit Windows 7. Stripe seems flummoxed by it and ngrok hasn't responded to a support request. After two days, my best guess is Satanic possession, but if Stripe or ngrok come through with a fix, I'll post it here.

**Comment by khannaabhi** [comment]
If you are using Virtual Box, there is a chance port only got exposed in VM. Go to VM VirtualBox and do port forwarding.
